src/dlnodesession.cpp

```diff
@@ -0,0 +1,294 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "dlnodesession.hpp"
+
+#include <map>
+#include <string>
+
+#include "logging.hpp"
+#include "modelinstance.hpp"
+#include "modelinstanceunloadguard.hpp"
+#include "modelmanager.hpp"
+#include "nodeinputhandler.hpp"
+#include "nodeoutputhandler.hpp"
+#include "nodestreamidguard.hpp"
+#include "tensorinfo.hpp"
+
+namespace ovms {
+DLNodeSession::DLNodeSession(const NodeSessionMetadata& metadata, const std::string& nodeName, uint32_t inputsCount, const ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+    NodeSession(metadata, nodeName, inputsCount),
+    modelManager(manager),
+    modelName(modelName),
+    modelVersion(modelVersion) {}
+
+DLNodeSession::DLNodeSession(const NodeSessionMetadata&& metadata, const std::string& nodeName, uint32_t inputsCount, const ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+    NodeSession(std::move(metadata), nodeName, inputsCount),
+    modelManager(manager),
+    modelName(modelName),
+    modelVersion(modelVersion) {}
+
+DLNodeSession::~DLNodeSession() = default;
+
+void DLNodeSession::clearInputs() {
+    this->inputHandler->clearInputs();
+}
+
+ModelInstance& DLNodeSession::getModelInstance() {
+    return *this->model;
+}
+
+InferenceEngine::InferRequest& DLNodeSession::getInferRequest(const uint microseconds) {
+    auto& inferRequestsQueue = this->model->getInferRequestsQueue();
+    auto streamIdOpt = this->nodeStreamIdGuard->tryGetId(microseconds);
+    if (!streamIdOpt) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Failed to get streamId on already executed node: {} session: {}", getName(), getSessionKey());
+        throw std::logic_error("Stream id is empty on already executed node");
+    }
+    return inferRequestsQueue.getInferRequest(streamIdOpt.value());
+}
+Status DLNodeSession::requestExecuteRequiredResources() {
+    Status status = StatusCode::OK;
+    ModelManager& manager = const_cast<ModelManager&>(modelManager);
```

Comment:
Can we make `getModelInstance` const? If not, I think we should pass non const reference of ModelManager to session.

Reply:
This way we could avoid const_cast.

Reply:
Yeah this was shortcut to not handle this issue at this point. To focus on the task. Will take a look at this.

Reply:
Yes - shouldn't pass const ModelManager to DLNodeSession.